lib/TableGen/Operations.cpp

```diff
@@ -159,8 +160,20 @@ unsigned OperationBase::getNumFullArguments() const {
 
 void OperationBase::emitArgumentAccessorDeclarations(llvm::raw_ostream &out,
                                                      FmtContext &fmt) const {
-  for (const auto &arg : m_arguments) {
+  std::map<std::string, uint32_t> argIndexMap;
```

Comment:
Can this be a simple vector from 0 to `m_arguments.size()`?
Apart from being simpler, that should also fix the nondeterministic iteration order when printing the enum variants.

Reply:
It cannot be `m_arguments.size()` since otherwise we reserve memory for the non-vararg arguments in the dialect op definition, leading to empty enum values, but otherwise it's a good idea, thanks.